src/refspec.c

```diff
@@ -160,6 +162,27 @@ void git_refspec__free(git_refspec *refspec)
 	memset(refspec, 0x0, sizeof(git_refspec));
 }
 
+int git_refspec_parse(git_refspec **refspec, const char *input, int is_fetch) {
+	git_refspec *new_refspec;
+	assert(refspec && input);
+
+	new_refspec = git__malloc(sizeof(git_refspec));
+	GITERR_CHECK_ALLOC(new_refspec);
+
+	if (!git_refspec__parse(new_refspec, input, is_fetch)) {
```

Comment:
You shouldn't pass `is_fetch` as, as `git_refspe__parse` accepts a bool. So instead of moving `is_fetch = !!is_fetch` into `git_refspec__parse`, you should have that here.

Reply:
@tiennou mentioned to move that line away. But I can move it back here.

Reply:
Sorry, coding style is always a soft thing and different people have different views :P

Reply:
On another note: you could also just move it into the `git_refspec__parse` line, without first assigning it to itself.